srv/ResetSimulation.srv

```diff
@@ -0,0 +1,7 @@
+# Reset the simulation to the start, including the entire scene and the simulation time.
```

Comment:
In Gazebo classic, there were multiple types of reset.

1. Reset poses/velocities but not time.
2. Reset everything including time.

But none of the de-spawned anything.

So I suggest adding a bitfield or similar structure where the user could specify the features that should be reset.

Reply:
Makes sense, I would also expect simulators to inform the users what type of resets are supported in the status_message

Reply:
I wouldn't return this kind of metadata in status_message (with each reset call). What about adding a general metadata message that would identify the features supported by the simulator?

Reply:
I think it should be part of this PR.

Things lets collect the ideas on its content. From the top of my head:

- reset support (and individual reset modes)
- single stepping support
- multi-stepping support
- pausing support (probably all sims will have it, but why not provide this info)
- dimensionality: 2D/3D
- dynamic spawning of entities (some sims might only work with what was specified in the world file)
- deleting entities
- support for links
- support for joints
- supported shapes (for visual/collisions)
- support for collision checking
- support for dynamics (acceleration, torque)

Inspiration can be collected in the simulation features of Gazebo:

https://github.com/gazebosim/gz-physics/blob/gz-physics8/dartsim/src/EntityManagementFeatures.hh
https://github.com/gazebosim/gz-physics/blob/gz-physics8/dartsim/src/FreeGroupFeatures.hh
https://github.com/gazebosim/gz-physics/blob/gz-physics8/dartsim/src/JointFeatures.hh
https://github.com/gazebosim/gz-physics/blob/gz-physics8/dartsim/src/KinematicsFeatures.hh
https://github.com/gazebosim/gz-physics/blob/gz-physics8/dartsim/src/LinkFeatures.hh
https://github.com/gazebosim/gz-physics/blob/gz-physics8/dartsim/src/ShapeFeatures.hh
https://github.com/gazebosim/gz-physics/blob/gz-physics8/dartsim/src/SimulationFeatures.hh
https://github.com/gazebosim/gz-physics/blob/gz-physics8/dartsim/src/WorldFeatures.hh

Reply:
I see that in some cases what you mean extends beyond support for `simulation_interfaces` services. 
While it makes sense to me to have a list of simulator capabilities, there is a plethora of possible features and multiple level of details, including sensor / controller supports. Maybe there can be an agree on some list of top level features. Maybe the best way to go is to keep it two separate bins:
- Support for `simulation_interface` services, including yes/no for services and then details on optional fields.
- Support for simulation features, as a list of strings, with text meant for the user.
I am interested in options here.

Reply:
I'm not a fan of text for the user. The use is very limited and it could be better substituted by a script that translates machine-readable values to user text.